docs/proposal/kmesh_support_encrypt.md

```diff
@@ -0,0 +1,181 @@
+--- 
+title: Kmesh支持节点之间进行数据加密
+authors:
+- "@bitcoffeeiux"
+reviews:
+-
+approves:
+-
+
+create-date: 2024-10-10
+
+---
+
+## 1.背景
+
+随着网络安全威胁的增加，未经加密的数据在传输过程中容易被黑客或第三方监听、截取甚至篡改，导致敏感信息泄露。为了解决上述安全风险，Kmesh计划引入节点数据加密模式，为节点之间的通信流量进行加密，消除通讯过程中的安全风险。
+
+## 2.使用场景
+
+由Kmesh进行代理的节点，在数据从应用发出时，路由到指定的网络接口设备进行加密处理后再经过网络发送给对端。对端由特定的接口接收到数据后进行解密，上送给对应的服务应用。
+
+![](./pics/p2p_encryption.png)
+
+## 3.IPsec简介
+
+IPsec是一个保护IP层通信的安全保密架构，是一个协议簇，通过对IP协议的分组进行加密和认证来保护IP协议的网络传输协议簇。运行在OSI模型的第三层（Internet Protocol，IP层），在VPN（virtual private networks）应用很广泛。
+有关IPsec的更多描述请参阅[什么是IPsec](https://info.support.huawei.com/info-finder/encyclopedia/zh/IPsec.html)
+
+## 4.Kmesh集成IPSec作为节点之间的加密工具
+
+Kmesh仅使用IPSec的加密功能，IPSec的密钥由用户设置在K8s后，传递至Kmesh进行管理并设置到IPSec中，保证IPSec的正常通信。整体架构图如下：
+
+![](./pics/Kmesh_ipsec_workload.png)
+
+### 4.1 用户设置IPSec密钥
+
+用户通过kubectl向K8s中设置名称为Kmesh-ipsec-keys的secret类型资源，资源中有如下信息：
+
+    [spi] aead-algo aead-key icv-len
+
+该资源中包含有ipsec使用的spi编号，ipsec使用的密钥算法，ipsec使用的预共享密钥以及预共享秘钥长度
+
+### 4.2 CRD设计
+
+Kmesh使能ipsec时，需要精细化控制ipsec数据加密行为，这其中要求Kmesh具有节点之间的信息同步机制。当前主要场景基于云原生业务场景，信息同步机制基于K8s集群api-server构建，依赖Kmesh自定义结构体来完成数据存储。
+
+CRD数据结构定义如下：
```

Comment:
The meaning of each field in crd can be illustrated from the contents of type.go